utils/lib/rr_graph/cyclic_to_acyclic.py

```diff
@@ -0,0 +1,201 @@
+class CycleSeas:
+    """
+    Contains the cycleseas that have been found in the graph so far.
+
+    A cyclesea is a the set of nodes contained in intersecting cycles so
+    that all nodes are accessible from one another.
+    """
+
+    def __init__(self):
+        # A dictionry that maps cyclesea_id to the set of nodes.
+        self.cycleseas = {}
+        # A dictionary that maps a node_id to the cyclesea index that it is present in.
```

Comment:
FYI We use NumPy style docstrings. These would be good to be part of that.

http://www.sphinx-doc.org/en/stable/ext/example_numpy.html?highlight=interface
https://numpydoc.readthedocs.io/en/latest/